test/suites/subnet/vnet_subnet_id_test.go

```diff
@@ -0,0 +1,140 @@
+	var dep *appsv1.Deployment
+	var selector labels.Selector
+	var numPods int
+
+	BeforeEach(func() {
+		numPods = 1
+		dep = test.Deployment(test.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "my-app"},
+				},
+			},
+		})
+		selector = labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+	})
+	It("should disrupt nodes that drifted due to VNETSubnetID", func() {
+		env.ExpectCreated(dep, nodeClass, nodePool)
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+		By("expect created node count to be 1")
+		env.ExpectCreatedNodeCount("==", 1)
+		nodeClaim := env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+
+		subnetName := "test-subnet"
+		subnet := &armnetwork.Subnet{
+			Name: lo.ToPtr(subnetName),
+			Properties: &armnetwork.SubnetPropertiesFormat{
+				AddressPrefix: lo.ToPtr("10.225.0.0/16"),
+			},
+		}
+		vnet := env.GetClusterVNET()
+		env.ExpectCreatedSubnet(lo.FromPtr(vnet.Name), subnet)
+		nodeClass.Spec.VNETSubnetID = subnet.ID // Should be populated by the Expect call above
+		env.ExpectCreatedOrUpdated(nodeClass)
+		env.EventuallyExpectDrifted(nodeClaim)
+	})
+	It("should allocate node in NodeClass subnet", func() {
+		subnetName := "test-subnet"
+		subnet := &armnetwork.Subnet{
+			Name: lo.ToPtr(subnetName),
+			Properties: &armnetwork.SubnetPropertiesFormat{
+				AddressPrefix: lo.ToPtr("10.225.0.0/16"),
+			},
+		}
+		vnet := env.GetClusterVNET()
+		env.ExpectCreatedSubnet(lo.FromPtr(vnet.Name), subnet)
+		nodeClass.Spec.VNETSubnetID = subnet.ID // Should be populated by the Expect call above
+
+		env.ExpectCreated(nodeClass, nodePool, dep)
+
+		env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+		nodes := env.EventuallyExpectCreatedNodeCount("==", 1)
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+
+		vm := env.GetVM(nodes[0].Name)
+		Expect(vm.Properties).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces).To(HaveLen(1))
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces[0].ID).ToNot(BeNil())
+		nicID, err := arm.ParseResourceID(*vm.Properties.NetworkProfile.NetworkInterfaces[0].ID)
+		Expect(err).ToNot(HaveOccurred())
+
+		// The NIC should have the right subnet
+		nic := env.GetNetworkInterface(nicID.Name)
+		Expect(nic.Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations).To(HaveLen(1))
+		Expect(nic.Properties.IPConfigurations[0].Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet.ID).To(Equal(subnet.ID))
+
+		// The NIC should have the right NSG
+		Expect(nic.Properties.NetworkSecurityGroup).ToNot(BeNil())
+		Expect(nic.Properties.NetworkSecurityGroup.ID).ToNot(BeNil())
+		Expect(*nic.Properties.NetworkSecurityGroup.ID).To(MatchRegexp(`aks-agentpool-\d{8}-nsg`))
```

Comment:
Found this other reference:
https://github.com/Azure/karpenter-provider-azure/blob/e02a20cdd65f5032848532891b028fd841da585a/pkg/providers/networksecuritygroup/networksecuritygroup.go#L37

Should it be case agnostic here? `(?i)^` prefix.